backend/src/layer/data/mail/mail_sender.rs

```diff
@@ -0,0 +1,183 @@
+        let mailer = transport_builder
+            .port(config.smtp_port)
+            .credentials(creds)
+            .build();
+        Ok(Self { config, mailer })
+    }
+
+    fn try_notify_admin_image_report(&self, info: &ImageReportInfo) -> MailResult<()> {
+        let sender = self.get_sender()?;
+        let reciever = self.get_receiver()?;
+        let report = Self::get_report(info);
+        let email = Message::builder()
+            .from(sender)
+            .to(reciever)
+            .subject(MAIL_SUBJECT)
+            .body(report)?;
+        self.mailer.send(&email)?;
+        info!(
+            "Email sent successfully for image with id {} at {}",
+            info.image_id, info.image_link
+        );
+        Ok(())
+    }
+
+    fn get_sender(&self) -> MailResult<Mailbox> {
+        let address = self.config.username.parse::<Address>()?;
+        Ok(Mailbox::new(Some(SENDER_NAME.to_string()), address))
+    }
+
+    fn get_receiver(&self) -> MailResult<Mailbox> {
+        let address = self.config.admin_email_address.parse::<Address>()?;
+        Ok(Mailbox::new(Some(RECEIVER_NAME.to_string()), address))
+    }
+
+    fn get_report(info: &ImageReportInfo) -> String {
+        let info_array_map: [(&str, &dyn ToString); 8] = [
+            ("image_link", &info.image_link),
+            ("image_id", &info.image_id),
+            ("report_count", &info.report_count),
+            ("reason", &info.reason),
+            ("image_got_hidden", &info.image_got_hidden),
+            ("positive_rating_count", &info.positive_rating_count),
+            ("negative_rating_count", &info.negative_rating_count),
+            ("get_image_rank", &info.get_image_rank),
+        ];
+
+        let info_map = info_array_map
+            .into_iter()
+            .map(|(k, v)| (k, v.to_string()))
+            .collect::<Vec<_>>();
+        let info_map = info_map.iter().map(|(k, v)| (*k, v.as_str())).collect();
+
+        Template::new(REPORT_TEMPLATE).render(&info_map)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    #![allow(clippy::unwrap_used)]
+    use crate::{
+        interface::admin_notification::ImageReportInfo, layer::data::mail::mail_info::MailInfo,
+        layer::data::mail::mail_sender::MailSender, util::Uuid,
+    };
+    use dotenvy;
+    use std::env::{self, VarError};
+
+    const SMTP_SERVER_ENV_NAME: &str = "SMTP_SERVER";
+    const SMTP_PORT_ENV_NAME: &str = "SMTP_PORT";
+    const SMTP_USERNAME_ENV_NAME: &str = "SMTP_USERNAME";
+    const SMTP_PASSWORD_ENV_NAME: &str = "SMTP_PASSWORD";
+    const ADMIN_EMAIL_ENV_NAME: &str = "ADMIN_EMAIL";
+
+    #[tokio::test]
+    async fn test_get_report() {
+        let info = get_report_info();
+        let report = MailSender::get_report(&info).replace("\r\n", "\n");
+        let expected = format!("The image at the url {}\nwith the id {}\nwas reported {} times.\nReason: {}\nImage automatically hidden: {}\n\nAdditional Data:\nPositive ratings: {}\nNegative ratings: {}\nRank: {}",
```

Comment:
that was not what i meant with testing the formating.
If there is no way to get an error from the template render method, you could check whether any pattern string using `{{ ...}}` (or whatever its syntax was) sill remains in the format (maybe if you missspelled an argument in the template file).
We dont want to change the test if we change the template.